rpcs3/Emu/Cell/Modules/cellSaveData.cpp

```diff
@@ -609,7 +609,12 @@ static NEVER_INLINE error_code savedata_op(ppu_thread& ppu, u32 operation, u32 v
 		listGet->dirList.set(setBuf->buf.addr());
 		std::memset(listGet->reserved, 0, sizeof(listGet->reserved));
 
-		const auto prefix_list = fmt::split(setList->dirNamePrefix.get_ptr(), {"|"});
+		auto prefix_list = fmt::split(setList->dirNamePrefix.get_ptr(), {"|"});
+
+		//if prefix_list is empty game wants to check all savedata
+		if (prefix_list.empty() && (operation == SAVEDATA_OP_LIST_LOAD || operation == SAVEDATA_OP_FIXED_LOAD)) {
```

Comment:
Another issue is that there might not be any relevance to the operation.
Why would the special case only apply to these 2 particular operations and not for all other operations that use the dirNamePrefix ?

Reply:
Because I don't understand what is the purpose of restricting operation only to specified prefix. I think that this way I disallow operations like writing to directory not owned by particular game. But probably all *_LOAD operations should be allowed ?

Reply:
There's nothing to understand, only mimic what the ps3 does.

Reply:
Do You mean that it is possible that the game will not work on real ps3 with current firmware ?

Reply:
From a developer perspective: either empty prefix works and lists all directories.. or the prefix string comes from somewhere else and *that* place is broken.

Reply:
No. He means that we should only do exactly what the PS3 does.
for example, if the dirNameString is empty, you'd need to find out which algorithm the PS3 uses to list folders for each of the operations, and then mimic it in our code

Reply:
Ah, I see, thanks.